ch360/fileredactor.go

```diff
@@ -0,0 +1,55 @@
+package ch360
+
+import (
+	"context"
+	"github.com/CloudHub360/ch360.go/ch360/request"
+	"io"
+)
+
+type FileRedactor struct {
+	docCreator   DocumentCreator
+	docExtractor DocumentExtractor
+	docRedactor  DocumentRedactor
+	docDeleter   DocumentDeleter
+}
+
+func NewFileRedactor(creator DocumentCreator, extractor DocumentExtractor,
+	redactor DocumentRedactor,
+	deleter DocumentDeleter) *FileRedactor {
+	return &FileRedactor{
+		docCreator:   creator,
+		docExtractor: extractor,
+		docRedactor:  redactor,
+		docDeleter:   deleter,
+	}
+}
+
+func (f *FileRedactor) Redact(ctx context.Context, fileContents io.Reader,
+	extractorName string) (io.ReadCloser, error) {
+
+	// Use a different context here so we don't cancel this req on ctrl-c. We need
```

Comment:
💪 I think we have this comment littered across the various command implementations. Could `docCreator.Create()` use the `Background` context by default so we don't have to keep duplicating this code and the related justification.